conbench/api/benchmarks.py

```diff
@@ -50,6 +51,33 @@ def get(self, benchmark_id):
         set_z_scores([benchmark_result])
         return self.serializer.one.dump(benchmark_result)
 
+    @flask_login.login_required
+    def put(self, benchmark_id):
+        """
+        ---
+        description: Edit a benchmark.
+        responses:
+            "200": "BenchmarkEntity"
+            "401": "401"
+            "404": "404"
+        parameters:
+          - name: benchmark_id
+            in: path
+            schema:
+                type: string
+        requestBody:
+            content:
+                application/json:
+                    schema: BenchmarkUpdate
+        tags:
+          - Benchmarks
+        """
+        benchmark_result = self._get(benchmark_id)
+        data = self.validate_benchmark(self.schema.update)
+        benchmark_result.update(data)
+        set_z_scores([benchmark_result])
```

Comment:
This PR does not yet exclude changes from the z-score calculation at all, right? If so, this line probably shouldn't be here since at this point it's at best re-calculating the same numbers as were calculated the first time (or if we've got bugs or we have an odd order of execution, it's changing the z-score numbers which could be incredibly confusing if you have a github alert saying there's some big regression, but then you click through and all is fine).

Reply:
Yep, it's correct that I haven't changed any calculations affecting z-score in this PR! But here's why I think this line is still okay:
- This line doesn't actually change anything in the database or the webapp; it just adds a `"z_score": X` field to the API response of the PUT request.
- We haven't changed any z-score calculations yet, so when you GET /api/benchmarks/<id> you'll always get a z-score that ignores any step changes.
- I think it's best to have the PUT response return the same schema as the GET response, and further I think it's better to have the z-score field return the same thing too (instead of nullifying out z-score for PUT).

Reply:
Aaah got it. This isn't something we should do here, but perhaps `set_z_scores` should have a more informative name `add_z_scores` `calculate_z_scores`? that make it a bit more obvious it's not persisting these anywhere